src/helpers/analytics.coffee

```diff
@@ -0,0 +1,64 @@
+_ = require 'underscore'
+Router = require 'react-router'
+{HistoryLocation} = require 'react-router'
+DestinationHelper = require '../helpers/routes-and-destinations'
+
+# generate custom event data for routes
+Events =
+  viewTaskStep: ({courseId}) ->
+    # track all work done on a course
+    Analytics.sendEvent 'Course', 'Work', label: courseId
+
+  viewStudentDashboard: ({courseId}) ->
+    # compare activity between courses
+    Analytics.sendEvent 'Student', 'Dashboard', label: courseId
+
+
+# Translators convert a url like '/foo/bar/123/baz/1' into a simplified one like just '/foo/bar'
+Translators =
```

Comment:
Is there a way to get the route href used to build the router (I think `react-router` allows you to build a router from JSON instead of components) and either use it directly or strip out the `:****`? hmm, or use the route name (`viewGuide`) instead of the URL?

That way, when we add more routes we do not have to update this file too?